terraform/ecs_ec2/daemon/main.tf

```diff
@@ -243,5 +245,20 @@ resource "null_resource" "validator" {
       go test ${var.test_dir} -timeout 0 -computeType=ECS -ecsLaunchType=EC2 -ecsDeploymentStrategy=DAEMON -cwagentConfigSsmParamName=${local.cwagent_config_ssm_param_name} -clusterArn=${aws_ecs_cluster.cluster.arn} -cwagentECSServiceName=${aws_ecs_service.cwagent_service.name} -v
     EOT
   }
-  depends_on = [aws_ecs_service.cwagent_service, aws_ecs_service.extra_apps_service]
+  depends_on = [aws_ecs_service.cwagent_service, aws_ecs_service.extra_apps_service, null_resource.disable_metadata]
 }
+
+resource "null_resource" "disable_metadata" {
+
+  provisioner "local-exec" {
+    command = <<-EOT
+      echo Sleep for 30 seconds to allow instance to be attached to the cluster
```

Comment:
When metadata is turned off immediately, ECS fails to attach the instance to the task so we will always fail.